airflow/providers/http/hooks/http.py

```diff
@@ -385,8 +387,9 @@ async def run(
             for attempt in range(1, 1 + self.retry_limit):
                 response = await request_func(
                     url,
-                    json=data if self.method in ("POST", "PUT", "PATCH") else None,
                     params=data if self.method == "GET" else None,
```

Comment:
maybe this should be changed too? this seems fishy also.

maybe this is too much change for folks but .... it doesn't seem like a good API to shuffle things around like this.  i.e. send `data` to params or data or json depending on this logic.  it would seem that it would be better to just forward everyting blindly.  @uranusjr @jedcunningham any thoughts?